jobs/build-kola-containers.Jenkinsfile

```
@@ -0,0 +1,221 @@
+            userRemoteConfigs: [[url: params.COREOS_ASSEMBLER_GIT_URL]],
+            extensions: [[$class: 'CloneOption',
+                          noTags: true,
+                          reference: '',
+                          shallow: true]]
+        ]
+    )
+
+    gitref = params.COREOS_ASSEMBLER_GIT_REF
+    def output = shwrapCapture("git rev-parse HEAD")
+    commit = output.substring(0,40)
+    shortcommit = commit.substring(0,7)
+
+    // gather the context folders list
+    def path = params.PATH_TO_CONTEXTS
+    contexts = shwrapCapture("""
+        cd ${path}
+        find . -maxdepth 1 -mindepth 1 -type d -exec basename {} \\;
+    """).trim().split("\n")
+
+    if ( ! params.FORCE ) {
+        // get the git commit ref for the last built container
+        def previous_ref = shWrapCapture("""
+            skopeo inspect --no-creds docker://${params.CONTAINER_REGISTRY_ORG}/${contexts[0]}:latest \
+            | jq -r '.Labels.org.opencontainers.image.revision'
+        """)
+
+        // Check for changes in tests/containers/* since the last build
+        // If none, no need to run this
+        // not using shWrapCapture here because we want the exit code 
+        def changeset = sh(returnStatus: true, script:"git diff --quiet ${previous_ref} -- ${path}")
+        changes = ( changeset == 1 )
+    } else {
+    changes = true
+  }
+}
+
+if ( !changes ) {
+    currentBuild.result = 'SUCCESS'
+    currentBuild.description = "[${gitref}@${shortcommit}] No changes in ${path}. Skiped the build."
+    return
+}
+
+currentBuild.description = "[${gitref}@${shortcommit}] Waiting"
+
+// Get the list of requested architectures to build for
+def basearches = params.ARCHES.split() as Set
+// and  the list of images to build
+def imageNames = contexts as Set
+
+lock(resource: "build-containers") {
+    cosaPod(image: params.COREOS_ASSEMBLER_IMAGE,
+            memory: "512Mi", kvm: false,
+            serviceAccount: "jenkins") {
+    timeout(time: 60, unit: 'MINUTES') {
+    try {
+
+        currentBuild.description = "[${gitref}@${shortcommit}] Running"
+
+        // By default we will allow re-using cache layers for one day.
+        // This is mostly so we can prevent re-downloading the RPMS
+        // and repo metadata and over again in a given day for successive
+        // builds.
+        def cacheTTL = "24h"
+        def force = ""
+        if (params.FORCE) {
+            force = '--force'
+            // Also set cacheTTL to 0.1s to allow users an escape hatch
+            // to force no cache layer usage.
+            cacheTTL = "0.1s"
+        }
+
+        withCredentials([file(credentialsId: 'cosa-push-registry-secret', variable: 'REGISTRY_SECRET')]) {
+            stage('Build Containers') {
+                parallel basearches.collectEntries{arch -> [arch, {
+                    for (imageName in imageNames) {
+                        def dirname = params.PATH_TO_CONTEXTS + imageName
```

```suggestion
                        def dirname = "${params.PATH_TO_CONTEXTS}/${imageName}"
```

So that we don't rely on the parameter always having a trailing slash.